crates/proof-of-sql/src/base/database/owned_table.rs

```diff
@@ -106,3 +106,102 @@ impl<S: Scalar> core::ops::Index<&str> for OwnedTable<S> {
             .unwrap()
     }
 }
+
+impl<'a, S: Scalar> From<&Table<'a, S>> for OwnedTable<S> {
+    fn from(value: &Table<'a, S>) -> Self {
+        match OwnedTable::try_from_iter(
+            value
+                .inner_table()
+                .iter()
+                .map(|(name, column)| (*name, OwnedColumn::from(column))),
+        ) {
+            Ok(table) => table,
+            Err(OwnedTableError::ColumnLengthMismatch) => {
+                panic!("Tables should not have columns with differing lengths")
+            }
+        }
+    }
+}
+
+impl<'a, S: Scalar> From<Table<'a, S>> for OwnedTable<S> {
+    fn from(value: Table<'a, S>) -> Self {
+        match OwnedTable::try_from_iter(
```

Comment:
We can just reuse the conversion above.

Reply:
This implementation moves rather than copies the column names. The other one would need to copy (clone when we use Ident)